tensorflow/python/ops/gradients_impl.py

```diff
@@ -1010,26 +1008,24 @@ def hessians(ys, xs, name="hessians", colocate_gradients_with_ops=False,
   hessians = []
   _gradients = gradients(ys, xs, **kwargs)
   for i, _gradient, x in zip(range(len(xs)), _gradients, xs):
-    # Ensure that x is a vector.
-    check_rank = check_ops.assert_rank(
-      x, 1, message='Cannot compute Hessian because element %d of `xs` does '
-      'not have rank one.' % i
+    # change shape to one-dimension without graph branching
+    _gradient = array_ops.reshape(_gradient, [-1])
+    
+    # Declare an iterator and tensor array loop variables for the gradients.
+    n = array_ops.size(x)
+    loop_vars = [
+      array_ops.constant(0, dtypes.int32),
+      tensor_array_ops.TensorArray(x.dtype, n)
+    ]
+    # Iterate over all elements of the gradient and compute second order
+    # derivatives.
+    _, hessian = control_flow_ops.while_loop(
+        lambda j, _: j < n,
+        lambda j, result: (j + 1,
+                           result.write(j, gradients(_gradient[j], x)[0])),
+        loop_vars
     )
-    with ops.control_dependencies([check_rank]):
-      # Declare an iterator and tensor array loop variables for the gradients.
-      n = array_ops.size(x)
-      loop_vars = [
-        array_ops.constant(0, dtypes.int32),
-        tensor_array_ops.TensorArray(x.dtype, n)
-      ]
-      # Iterate over all elements of the gradient and compute second order
-      # derivatives.
-      _, hessian = control_flow_ops.while_loop(
-          lambda j, _: j < n,
-          lambda j, result: (j + 1,
-                             result.write(j, gradients(_gradient[j], x)[0])),
-          loop_vars
-      )
-
-      hessians.append(hessian.stack())
+
+    _shape = x.shape.as_list()
```

Comment:
Let's use `array_ops.shape` here and then `array_ops.concat` in the line below such that we do not require the shape to be known a priori.